akka-projection-jdbc/src/main/scala/akka/projection/jdbc/internal/Dialect.scala

```diff
@@ -45,38 +88,113 @@ private[jdbc] object DialectDefaults {
-      "MERGEABLE" BOOLEAN NOT NULL,
-      "LAST_UPDATED" TIMESTAMP(9) WITH TIME ZONE NOT NULL
+  override val createTableStatements: immutable.Seq[String] = DialectDefaults.createTableStatement(table)
+
+  override val readOffsetQuery: String = DialectDefaults.readOffsetQuery(table)
+
+  override val clearOffsetStatement: String = DialectDefaults.clearOffsetStatement(table)
+
+  override def insertStatement(): String = DialectDefaults.insertStatement(table)
+
+  override def updateStatement(): String = DialectDefaults.updateStatement(table)
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[jdbc] case class MySQLDialect(schema: Option[String], tableName: String) extends Dialect {
+
+  def this(tableName: String) = this(None, tableName)
+
+  private val table = schema.map(s => s"$s.$tableName").getOrElse(tableName)
+  override val createTableStatements =
+    immutable.Seq(
+      s"""
+     CREATE TABLE IF NOT EXISTS $table (
+      PROJECTION_NAME VARCHAR(255) NOT NULL,
+      PROJECTION_KEY VARCHAR(255) NOT NULL,
+      OFFSET VARCHAR(255) NOT NULL,
+      MANIFEST VARCHAR(4) NOT NULL,
+      MERGEABLE BOOLEAN NOT NULL,
+      LAST_UPDATED TEXT NOT NULL
      );
-     """
+    """,
+      // create index
+      s"""CREATE INDEX PROJECTION_NAME_INDEX ON $table (PROJECTION_NAME);""",
+      // add primary key
+      s"""ALTER TABLE $table
+       ADD CONSTRAINT PK_PROJECTION_ID PRIMARY KEY(PROJECTION_NAME,PROJECTION_KEY);
+       """)
 
-  override val alterTableStatement =
-    s"""
-       ALTER TABLE "$table" 
-       ADD CONSTRAINT "PK_PROJECTION_ID" PRIMARY KEY("PROJECTION_NAME","PROJECTION_KEY");
-    """
+  override val readOffsetQuery: String =
+    removeQuotes(DialectDefaults.readOffsetQuery(table))
 
-  override val readOffsetQuery: String = DialectDefaults.readOffsetQuery(table)
+  override val clearOffsetStatement: String =
+    removeQuotes(DialectDefaults.clearOffsetStatement(table))
 
-  override val insertOrUpdateStatement = DialectDefaults.insertOrUpdateStatement(table)
+  override def insertStatement(): String =
+    removeQuotes(DialectDefaults.insertStatement(table))
+
+  override def updateStatement(): String =
+    removeQuotes(DialectDefaults.updateStatement(table))
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi
+private[jdbc] case class MSSQLServerDialect(schema: Option[String], tableName: String) extends Dialect {
+
+  def this(tableName: String) = this(None, tableName)
+
+  private val table = schema.map(s => s"$s.$tableName").getOrElse(tableName)
+  override val createTableStatements =
+    immutable.Seq(s"""
+      IF  NOT EXISTS (SELECT 1 FROM sys.objects WHERE object_id = OBJECT_ID(N'$table') AND type in (N'U'))
+      begin
+      create table "$table" (
+        "PROJECTION_NAME" VARCHAR(255) NOT NULL,
+        "PROJECTION_KEY" VARCHAR(255) NOT NULL,
+        "OFFSET" VARCHAR(255) NOT NULL,
+        "MANIFEST" VARCHAR(4) NOT NULL,
+        "MERGEABLE" BIT NOT NULL,
+        "LAST_UPDATED" DATETIME2 NOT NULL
```

Comment:
DATETIME2 is not the original type that Slick is using, it used DATETIMEOFFSET. 

The problem with the later is that get the double conversion:


Java instant:
`2020-06-24T10:47:21.918Z` 

SQL Timestamp
`2020-06-24 12:47:21.918` 

SQL Timestamp when read back
`2020-06-24 14:47:21.9180000`

Java instant 
`2020-06-24 12:47:21.918`

We probably didn't see it because we only run tests with H2 for Slick. This needs to be fixed there.